contracts/tonomy/src/tonomy.cpp

```diff
@@ -212,6 +212,67 @@ namespace tonomysystem
          row.version = 1; });
    }
 
+   void newappadmin(
+      name user_name,  
```

Comment:
i added username and change random_name to username so we can identify which user added these apps.. or if we keep it random name how to see which user have which apps ?

Reply:
yes that is good. I would rename to `owner`.

we should also check that the `owner` account is a Person type

in newapp() and newappadmin(), Instead of using the `public_key` we use a delegated authorization so that the account is controlled by the owner (user_name). `public_key` arg should be deleted